search/linear_search.cpp

```diff
@@ -63,30 +67,42 @@ static void tests() {
  * @brief Main function
  * @returns 0 on exit
  */
-int main() {
+int main()
+{
     int mode = 0;
 
     std::cout << "Choose mode\n";
     std::cout << "Self-test mode (1), interactive mode (2): ";
 
     std::cin >> mode;
 
-    if (mode == 2) {
+    if (mode == 2)
+    {
         int size = 0;
-        std::cout << "\nEnter the size of the array: ";
+        std::cout << "\nEnter the size of the array [in range 1-30 ]: ";
         std::cin >> size;
 
-        while ((size <= 1) || (size >= 30)) {
-            std::cout << "Size cannot be less than zero. Please choose another value: ";
-            std::cin >> size;
+        while (size <= 0 || size > 30)
+        {
+            if (size <= 0)
+            {
+                std::cout << "Size cannot be less than zero. Please choose another value: ";
+                std::cin >> size;
+            }
+            if (size > 30)
+            {
+                std::cout << "Size entered more than 30. Please choose a lower value: ";
+                std::cin >> size;
+            }
```

Comment:
I really feel code getting brittle. Can not we rethink? And we have to take input anyways so input can be taken anyways should not be in if block.

Reply:
The second if condition is really not needed, but for the programmer's concern of taking size less than 30 that part of code has been added.